src/scene/shader-lib/chunks/lit/frag/clusteredLight.js

```diff
@@ -135,18 +139,38 @@ vec4 decodeClusterLowRange4Vec4(vec4 d0, vec4 d1, vec4 d2, vec4 d3) {
     );
 }
 
-vec4 sampleLightsTexture8(const ClusterLightData clusterLightData, float index) {
-    return texture2DLodEXT(lightsTexture8, vec2(index * lightsTextureInvSize.z, clusterLightData.lightV), 0.0);
-}
+#ifdef GL2
 
-vec4 sampleLightTextureF(const ClusterLightData clusterLightData, float index) {
-    return texture2DLodEXT(lightsTextureFloat, vec2(index * lightsTextureInvSize.x, clusterLightData.lightV), 0.0);
-}
+    vec4 sampleLightsTexture8(const ClusterLightData clusterLightData, int index) {
+        return texelFetch(lightsTexture8, ivec2(index, clusterLightData.lightIndex), 0);
+    }
+
+    vec4 sampleLightTextureF(const ClusterLightData clusterLightData, int index) {
+        return texelFetch(lightsTextureFloat, ivec2(index, clusterLightData.lightIndex), 0);
+    }
+
+#else
+
+    uniform vec4 lightsTextureInvSize;
```

Comment:
Would it be better to keep the uniforms in one place (at the top) and just #ifdef it there?

Reply:
yep done, thanks